python/kvikio/cufile.py

```diff
@@ -230,13 +229,13 @@ def write(self, buf, size: int = None, file_offset: int = 0, task_size=None) ->
 
         Notes
         -----
-        cuFile can only make use of GPUDirect Storage for writes that are
-        aligned to a page boundary. The GPU page size used by cuFile is 4kB, so
-        all reads must be at an offset that is a multiple of 4096 bytes. If the
-        desired `file_offset` is not a multiple of 4096 it is likely desirable
-        to round down to the nearest multiple of 4096 and discard any undesired
-        bytes from the resulting data. Similarly, it is optimal for `size` to
-        be a multiple of 4096 bytes.
+        When writing from device memory, cuFile can only make use of GDS for writes
```

Comment:
Wonder if we can make a fallback (for when GDS is not present) where it copies from device memory to pinned memory and then writes out. The allocation cost could be amortized away by reusing allocations from a pinned memory pool.

Reply:
We actually do this, updated the docs